packages/backend/src/db/storage/helpers/freeze-template.ts

```diff
@@ -0,0 +1,15 @@
+// Helper function to ensure templates cannot be modified
+export function deepFreeze<T>(object: T): T {
```

Comment:
can use type-fest's ReadonlyDeep 
https://github.com/sindresorhus/type-fest

Reply:
hmm i tried but i realised graphql expects a mutable `Template` type to be returned so its a little problematic cus need to convert back on the backend query, is it ok to just handroll this function cus its not too complex?